tests/core/engine_adapter/integration/docker/spark/Dockerfile

```diff
@@ -1,4 +1,6 @@
-FROM docker.io/bitnami/spark:3.5
+# Pinned to a specific version due to PySpark Pandas requiring distutils which dissappears in python 3.12
+# Should be fixed in v4 https://issues.apache.org/jira/browse/SPARK-45390
+FROM docker.io/bitnami/spark:3.5.2-debian-12-r0
```

Comment:
I assume that the Circle CI config is referenced from the main trunk rather than this branch. If not then please let me know!